internal/source/source.go

```diff
@@ -198,6 +217,9 @@ func (s *Source) Teardown(context.Context) error {
 		if err := s.iterator.Stop(); err != nil {
 			return fmt.Errorf("stop iterator: %w", err)
 		}
+
+		// closing nats connection
+		s.nc.Close()
```

Comment:
Maybe do this as part of `iterator.Stop()`? That way, the source will only be talking to the iterator only and doesn't need to go to the "lower level", i.e. the client?

Reply:
I do prefer to keep closing the connection [where we open it](https://github.com/conduitio-labs/conduit-connector-nats-jetstream/pull/68/files#diff-31f0dc97cfd1c9aaef8c514393a23f48a8fc8a6b2c34217a7b0d3e3d9edbd0cfR158). The iterator is responsible for managing the connection context and iterating the data.